src/services/wallet-service/wallet-service.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { WalletService, WALLET_STORAGE_KEY } from "./wallet-service";
+import { MockStorage } from "../../../test-config/mocks/MockStorage";
+import { MockMerkleTreeService } from "../../../test-config/mocks/MockMerkleTreeService";
+import { MockWalletService } from "../../../test-config/mocks/MockWalletService";
+import { Observable } from "rxjs/Observable";
+import 'rxjs/add/observable/of';
+import { IWallet } from "../../models/IWallet";
+import { MerkleTreeService } from "../merkle-tree-service/merkle-tree-service";
+
+describe("WalletService", () => {
+    let walletService: WalletService;
+    let storage: MockStorage;
+    let merkleTreeService: MockMerkleTreeService;
+    let mockedWalletService: MockWalletService;
+
+    beforeEach(() => {
+        storage = new MockStorage();
+        merkleTreeService = new MockMerkleTreeService();
+        mockedWalletService = new MockWalletService();
+        walletService = new WalletService(storage, <MerkleTreeService>merkleTreeService);
+    });
+
+    it("should return an empty array when there are no wallets saved", (done) => {
+        spyOn(storage, "get").and.returnValue(Promise.resolve());
+        walletService.getAll().then(data => {
+            expect(storage.get).toHaveBeenCalledWith(WALLET_STORAGE_KEY);
+            expect(data).toEqual([]);
+            done();
+        });
+    });
+
+    it("should return three specific wallets when retrieving the wallets from storage", (done) => {
+        mockedWalletService.getAll().then(data => {
+            spyOn(storage, "get").and.returnValue(Promise.resolve(data));
+            walletService.getAll().then(data => {
+                expect(data).toEqual(<any>[
+                    {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                        "type":"local",
+                        "name":"Biosta",
+                        "publicKey":
+                        "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
```

Comment:
These wallets are using an outdated definition. `encryptedPrivateKey` is replaced by `keyStore`. Does not really matter for this test but maybe update this to be in line with how wallets are truly defined?

Reply:
Updated with how the wallets are truly defined